### PR TITLE
Add Katarina's postprocessing of HF correlations

### PR DESCRIPTION
**Pull request by saganatt**
Ciao @ginnocen @vkucera 

these correspond to the task in https://github.com/AliceO2Group/O2Physics/pull/989 .
I copied Katarina's branch: https://github.com/Katka19/Run3Analysisvalidation/tree/HFcorr. I removed AnalysisResults_O2.root from git and cleaned the code with clang format.
Let me know what are the conventions esp. wrt. the config and JSON files. What default settings do you prefer?

### CODE REVIEW

File: codeHF/config_tasks.sh [hunk 1]
```diff
@@ -71,6 +74,7 @@ DOO2_TASK_CHIC=0    # hf-task-chic
 DOO2_TASK_LCK0SP=0  # hf-task-lc-tok0sp
 DOO2_TASK_XICC=0    # hf-task-xicc
 DOO2_TASK_BPLUS=0   # hf-task-bplus
+DOO2_TASK_HFCORR=1  # hf-task-hfcorrelations
```
Comment (vkucera) [inline codeHF/config_tasks.sh:77]: Move under `# Correlations`

File: codeHF/config_tasks.sh [hunk 2]
```diff
@@ -260,6 +264,9 @@ function MakeScriptO2 {
   [ "$ISALICE3" -eq 1 ] && SUFFIX_ALICE3="-alice3" || SUFFIX_ALICE3=""
 
   WORKFLOWS=""
+  # MC converter to version 001 of mcparticles
+  [ $DOO2_MCCONV -eq 1 ] && WORKFLOWS+=" o2-analysis-mc-converter"
+  [ $DOO2_FDDCONV -eq 1 ] && WORKFLOWS+=" o2-analysis-fdd-converter"
```
Comment (vkucera) [inline codeHF/config_tasks.sh:269]: Remove (already there)

File: codeHF/dpl-config_run3.json [hunk 3]
```diff
@@ -64,6 +74,7 @@
         "doCutQuality": "true",
         "useIsGlobalTrack": "false",
         "tpcNClsFound": "71",
+        "useIsGlobalTrack": "true",
```
Comment (vkucera) [inline codeHF/dpl-config_run3.json:77]: Remove

File: codeHF/config_input.sh [hunk 4]
```diff
@@ -75,4 +76,11 @@ case $INPUT_CASE in
     INPUT_FILES="AO2D.root"
     ISINPUTO2=1
     ISMC=1;;
+  10)
+    INPUT_LABEL="Run 2 converted data, pp 13.6 TeV, LHC17j period"
+    #INPUT_DIR="/data2/Run2samples/ConvertedData/LHC17o_date20220525/281961" # converted bad AO2Ds
```
Comment (vkucera) [inline codeHF/config_input.sh:81]: Is the commented line needed?

**Review by vkucera — CHANGES_REQUESTED** [review]

File: codeHF/dpl-config_run3.json [hunk 5]
```diff
@@ -46,21 +46,11 @@
...
-        "estV0M": "-1",
-        "estRun2SPDtks": "-1",
-        "estRun2SPDcls": "-1",
-        "estRun2CL0": "-1",
-        "estRun2CL1": "-1",
-        "ccdburl": "http://alice-ccdb.cern.ch",
-        "ccdbpath": "Centrality/Estimators",
-        "genname": ""
+        "isRun3": "false"
     },
```
Comment (vkucera) [inline codeHF/dpl-config_run3.json:54]: I think this should be kept as it is a new added configuration.

File: codeHF/dpl-config_run3.json [hunk 6]
```diff
@@ -961,176 +951,6 @@
         "selectionFlagLcbar": "0",
         "cutEtaCandMax": "-1"
     },
-    "task-hf-correlations": {
```
Comment (vkucera) [inline codeHF/dpl-config_run3.json]: This should also be kept.

**Review by vkucera — CHANGES_REQUESTED** [review]

File: codeHF/config_tasks.sh [hunk 7]
```diff
@@ -84,6 +84,7 @@ DOO2_D0D0BAR_MCGEN=0     # hf-correlator-d0d0bar-mc-gen
 DOO2_DPLUSDMINUS_DATA=0  # hf-correlator-dplusdminus
 DOO2_DPLUSDMINUS_MCREC=0 # hf-correlator-dplusdminus-mc-rec
 DOO2_DPLUSDMINUS_MCGEN=0 # hf-correlator-dplusdminus-mc-gen
+DOO2_D0=0  # hf-task-correlation-d0
```
Comment (vkucera) [inline codeHF/config_tasks.sh:87]: `DOO2_D0` is too generic. Use something like `DOO2_TASK_CORRD0`.

File: codeHF/config_tasks.sh [hunk 8]
```diff
@@ -314,6 +315,7 @@ function MakeScriptO2 {
   [ $DOO2_DPLUSDMINUS_DATA -eq 1 ] && WF_CORR="o2-analysis-hf-correlator-dplusdminus o2-analysis-hf-task-correlation-ddbar"
   [ $DOO2_DPLUSDMINUS_MCREC -eq 1 ] && WF_CORR="o2-analysis-hf-correlator-dplusdminus-mc-rec o2-analysis-hf-task-correlation-ddbar-mc-rec"
   [ $DOO2_DPLUSDMINUS_MCGEN -eq 1 ] && WF_CORR="o2-analysis-hf-correlator-dplusdminus-mc-gen o2-analysis-hf-task-correlation-ddbar-mc-gen"
+  [ $DOO2_D0 -eq 1 ] && WF_CORR="o2-analysis-hf-task-correlation-d0"
```
Comment (vkucera) [inline codeHF/config_tasks.sh:318]: Aren't `o2-analysis-hf-task-correlation-d0` and the tasks in `WF_CORR` independent?

**Review by vkucera — CHANGES_REQUESTED** [review]

File: codeHF/config_tasks.sh (hunk 8 above)
Comment (saganatt) [reply codeHF/config_tasks.sh]: I thought it is like a category for all correlations. I changed it back.

**Comment by vkucera** [comment]
> I am also thinking about a less generic name for the `FirstAnalysis` directory. Could it be, e.g., `D0CorrPostProcessing`?

The directory `FirstAnalysis` already exists and contains different kinds of scripts and macros.

File: codeHF/config_input.sh [hunk 9]
```diff
@@ -75,4 +75,10 @@ case $INPUT_CASE in
     INPUT_FILES="AO2D.root"
     ISINPUTO2=1
     ISMC=1;;
+  10)
+    INPUT_LABEL="Run 2 converted data, pp 13.6 TeV, LHC17j period"
+    INPUT_DIR="/data2/Run2samples/ConvertedData/LHC17j_date20220601/274671" # converted good AO2Ds
+    INPUT_FILES="AO2D.root"
+    INPUT_RUN=2
```
Comment (vkucera) [inline codeHF/config_input.sh:82]: `INPUT_RUN=2` is default. No need to put it here.

File: codeHF/config_input.sh [hunk 10]
```diff
@@ -75,4 +75,10 @@ case $INPUT_CASE in
     INPUT_FILES="AO2D.root"
     ISINPUTO2=1
     ISMC=1;;
+  10)
+    INPUT_LABEL="Run 2 converted data, pp 13.6 TeV, LHC17j period"
```
Comment (vkucera) [inline codeHF/config_input.sh:79]:
```suggestion
    INPUT_LABEL="Run 2, p-p 13 TeV LHC17j, (AO2D)"
```

**Review by vkucera — CHANGES_REQUESTED** [review]

**Comment by saganatt** [comment]
I found my mistake, the task name was wrong in JSON. Now the settings are picked correctly.
On the other hand, I get hadd errors on merging.
`Fatal in <TFileMerger::RecursiveRemove>: Output file of the TFile Merger (targeting AnalysisResults.root) has been deleted (likely due to a TTree larger than 100Gb)
`
The previous time I got such errors, they disappeared randomly, maybe some re-compilation helped. Now compiling with ninja or the whole aliBuild doesn't help. Do you maybe have some suggestions?

File: codeHF/dpl-config_run3.json [hunk 11]
```diff
@@ -153,23 +153,12 @@
         "processRun3": "false"
     },
     "tof-event-time": {
-        "param-file": "",
```
Comment (saganatt) [inline codeHF/dpl-config_run3.json:156]: This needs to be removed to stop crashing according to Nicolo: https://mattermost.web.cern.ch/alice/pl/srdmx9wupfgkdr1qy3rieu9pze

Comment (vkucera) [reply codeHF/dpl-config_run3.json:156]: Should be fixed in the most recent version.

File: codeHF/dpl-config_run3.json [hunk 12]
```diff
@@ -951,7 +940,7 @@
         "selectionFlagLcbar": "0",
         "cutEtaCandMax": "-1"
     },
-    "hf-task-correlation-d0": {
+    "task-correlation-d0": {
```
Comment (vkucera) [inline codeHF/dpl-config_run3.json:943]: The device name should start with HF, so `hf-task-correlation-d0` was correct and the O2Physics code should be modified accordingly.

**Pull request closed without merging**